File: ABC/abc008/abc008_3 3.py
#100点解法
N = int(input())
C = [int(input()) for _ in range(N)]

ans = 0
for i in range(N): #C[i]のコインが最後に表を向いている確率を調べる
    count = 0 #C[j](0≤j<N,j≠i)のうち、C[i]の約数の個数
    for j in range(N): #C[j]がC[i]の約数かどうかを調べる
        if C[i] % C[j] == 0 and i != j:
            count += 1
    ans += (count // 2 + 1) / (count + 1)

print(ans)

File: ABC/abc008/abc008_3 3.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    // 100点解法
    int N;
    cin >> N;
    vector<int> C(N);
    for (int i = 0; i < N; i++) cin >> C[i];
    
    double ans = 0;
    for (int i = 0; i < N; i++) {
        int count = 0; // C[j](0≤j<N,j≠i)のうち、C[i]の約数の個数
        for (int j = 0; j < N; j++) {
            if (C[i] % C[j] == 0 && i != j) count++;
        }

        ans += (count / 2 + 1.0) / (count + 1.0);
    }

    cout << fixed << setprecision(10) << ans << endl;
}